airbyte-ci/connectors/pipelines/pipelines/commands/groups/connectors.py

```diff
@@ -142,7 +123,13 @@ def test(
         main_logger.info("Skipping connectors tests for draft pull request.")
         sys.exit(0)
 
-    main_logger.info(f"Will run the test pipeline for the following connectors: {', '.join(ctx.obj['selected_connectors_names'])}")
+    if ctx.obj["select_modified_connectors"]:
+        ctx.obj["selected_connectors_and_files"] = {
+            **ctx.obj["selected_connectors_and_files"],
+            **get_modified_connectors_and_files(ctx.obj["modified_files"]),
+        }
```

Comment:
Can you help me understand what's going on here with the `**`? 

I think this means "merge `ctx.obj["selected_connectors_and_files"]` with `get_modified_connectors_and_files(ctx.obj["modified_files"])`, but shouldn't `select_modified_connectors` only have the result of `get_modified_connectors_and_files`? 

It seems like this is adding what you've selected with `--name` or other args with the modified connectors. But I would expect the `--name`, `--language`, etc to _filter_ the modified connectors. At least that was the way it was working before and is how I stopped java connectors from publishing here https://github.com/airbytehq/airbyte/pull/28344

Reply:
Yep `**` expands a dictionnary, in this context it indeeds merge two dictonaries into one. 
And yes you're right it slightly changes the logic. I'll try to revert to the original.
The expected logic is:
* Combination of same options like `--name`, `--language` etc. act as `OR`
* Combination of different options are `AND` `--language=python --language=java --release-stage=generally_available` should only test Python and java GA connectors.

I'll make sure to add a test for this.

Reply:
@alafanechere I think we're still `and`-ing modified and the other filters. Why does --modified not work like the others? 

For example `--modified --language=java` would select only the modified java connectors. Maybe I'm not reading this right...

Reply:
@pedroslopez You're right again 😮‍💨 I originally thought it was OK to `or` the modified connectors but for the publish use case is was cool to just bypass specific connector language.
I changed the code structure again to centralize the logic even more and added thorough unit tests.